lib/_nonstd.scm

```diff
@@ -2700,11 +2700,7 @@
                        (expand relpath
                                expanded-dir)))
                     ((##string=? instdir-name "execdir")
-                     (let ((exec-path (##os-executable-path)))
-                       (if (##fixnum? exec-path)
-                           (err exec-path)
-                           (expand relpath
-                                   (##path-directory exec-path)))))
+                     (expand-in-instdir relpath "bin"))
```

Comment:
This is plain wrong. It goes against the reason behind `~~execdir`, which is to locate the directory containing the executable when other related files are relative to that.

You have to understand that `gsi` and `gsc` are just 2 special programs that are built with Gambit. In general a program built with Gambit could be installed anywhere, not in the directory `~~bin` which is the directory where Gambit specific programs are installed, when they are installed in the usual way. But even `gsi` and `gsc` could be built in one location (for a "binary" distribution) and then installed on a user's machine to a user preferred location, so the `~~execdir` can be useful to allow Gambit specific files to be found. Specifically, if a Gambit installation has been `mv`ed to a different place and `gsi` is accessible in your `PATH` you can still execute it with `gsi -:~~=~~execdir/..` (which defines `~~` to be the parent directory of the exec dir, i.e. if the exec dir is `/foo/bar/bin` then `~~bin` will resolve to `/foo/bar/bin`, `~~lib` will resolve to `/foo/bar/lib`, etc).

Reply:
Reverted this one, so for now that clause will just raise an exception if `##os-executable-path` doesn't work.

Thank you for the clear explanation.  Especially the first sentence of the 2nd paragraph.  On some level I _knew_ that, but that wasn't how I was _thinking about_ the problem...if that makes sense?